core-backend/src/mock.rs

```diff
@@ -22,22 +22,54 @@ use crate::{
     },
     BackendExternalities,
 };
-use alloc::{collections::BTreeSet, vec, vec::Vec};
+use alloc::{collections::BTreeSet, rc::Rc, vec, vec::Vec};
 use codec::{Decode, Encode};
-use core::{cell::Cell, fmt, fmt::Debug};
+use core::{fmt, fmt::Debug};
 use gear_core::{
     costs::CostToken,
     env::{Externalities, PayloadSliceLock, UnlockPayloadBound},
     env_vars::{EnvVars, EnvVarsV1},
     gas::{ChargeError, CounterType, CountersOwner, GasAmount, GasCounter, GasLeft},
     ids::{MessageId, ProgramId, ReservationId},
-    memory::{Memory, MemoryError, MemoryInterval},
+    memory::{Memory, MemoryInterval},
     message::{HandlePacket, InitPacket, ReplyPacket},
     pages::{PageNumber, PageU32Size, WasmPage, WASM_PAGE_SIZE},
 };
 use gear_core_errors::{ReplyCode, SignalCode};
 use gear_lazy_pages_common::ProcessAccessError;
+use gear_sandbox::{default_executor::Store, AsContextExt, SandboxMemory};
 use gear_wasm_instrument::syscalls::SyscallName;
+use std::cell::RefCell;
+
+thread_local! {
+    static MEMORY_ACCESSES: RefCell<PreProcessMemoryAccesses> = const { RefCell::new(PreProcessMemoryAccesses::new()) };
+}
+
+#[derive(Debug)]
+pub struct PreProcessMemoryAccesses {
+    pub(crate) reads: Vec<MemoryInterval>,
+    pub(crate) writes: Vec<MemoryInterval>,
+}
+
+impl PreProcessMemoryAccesses {
```

Comment:
To consider. We can make CallerWrap::pre_process_memory_access(&mut self, ...) in order to skip this thread_local creation

Reply:
Could you explain how `CallerWrap`'s method may help to avoid creation?

Reply:
A, mistake. I mean append argument `&mut self` to Ext::pre_process_memory_access()